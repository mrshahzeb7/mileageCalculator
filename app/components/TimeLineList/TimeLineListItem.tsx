import React from 'react';
import {StyleSheet, Text, TextStyle, View, ViewStyle} from 'react-native';
import {icons} from '../../assets';
import {colors, iconSizes, spacing} from '../../theme';
import {Icon} from '../Icon/Icon';
import {responsiveHeight} from '../../utils';
import {TimeLineSectionItem, mileageUnit} from './timeLineListContent';

interface Props {
  item: TimeLineSectionItem;
  index: number;
}

export const timeLineListItemHeight = responsiveHeight(13);

export const TimeLineListItem: React.FC<Props> = ({item}) => {
  return (
    <View style={styles.container}>
      <View style={styles.leftWrapper}>
        <LeftIcon />
        <View style={styles.detailWrapper}>
          <Text style={styles.title}>Refueling</Text>
          <Text style={styles.date}>{item?.date}</Text>

          <Mileage
            value={item?.meterReading + ' ' + mileageUnit.milesPerLitre}
          />
        </View>
      </View>

      <Text style={styles.price}>$ {item?.cost}</Text>
    </View>
  );
};

const Mileage = ({value}: {value?: string}) => {
  return (
    <View style={styles.mileageContainer}>
      <Icon icon={icons.Meter} color={colors.textDim} size={iconSizes.tiny} />
      <Text style={styles.mileage}>{value}</Text>
    </View>
  );
};

const LeftIcon = () => {
  return (
    <Icon
      containerStyle={styles.iconContainer}
      size={iconSizes.tiny}
      icon={icons.Gas}
      color={colors.light}
    />
  );
};
const $row: ViewStyle = {
  flexDirection: 'row',
  alignItems: 'center',
};

const $textStyle: TextStyle = {
  color: colors.text,
};

const $dimTextStyle: TextStyle = {
  color: colors.textDim,
};
// styles

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: spacing.horizontal.small,
    justifyContent: 'space-between',
    height: timeLineListItemHeight,
    ...$row,
  },

  leftWrapper: {
    ...$row,
  },

  iconContainer: {
    backgroundColor: colors.primaryButton,
    height: iconSizes.large,
    width: iconSizes.large,
    borderRadius: iconSizes.large,
    alignItems: 'center',
    justifyContent: 'center',
  },
  detailWrapper: {
    marginLeft: spacing.horizontal.medium,
    marginTop: spacing.vertical.medium,
  },
  mileageContainer: {
    justifyContent: 'center',
    ...$row,
  },
  title: {
    ...$textStyle,
  },
  date: {
    marginVertical: spacing.vertical.tiny,
    ...$dimTextStyle,
  },
  mileage: {
    color: colors.textDim,
    marginLeft: spacing.horizontal.small,
    ...$dimTextStyle,
  },
  price: {
    ...$textStyle,
  },
});
