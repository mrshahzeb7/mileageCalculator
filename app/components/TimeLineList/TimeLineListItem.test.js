import 'react-native';
import React from 'react';
import {TimeLineListItem} from './TimeLineListItem';
import renderer from 'react-test-renderer';

describe('TimeLineListItem', () => {
  it(' TimeLineListItem renders correctly', () => {
    const tree = renderer.create(<TimeLineListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
