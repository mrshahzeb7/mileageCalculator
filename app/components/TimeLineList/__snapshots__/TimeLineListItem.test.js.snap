// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLineListItem  TimeLineListItem renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 173.42000000000002,
      "justifyContent": "space-between",
      "paddingHorizontal": 15,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#93b2fd",
          "borderRadius": 81.07922223219569,
          "height": 81.07922223219569,
          "justifyContent": "center",
          "width": 81.07922223219569,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../app/assets/icons/general/gas.png",
          }
        }
        style={
          [
            {
              "resizeMode": "contain",
            },
            {
              "tintColor": "#ffffff",
            },
            {
              "height": 30.13699392404255,
              "width": 30.13699392404255,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        {
          "marginLeft": 37.5,
          "marginTop": 66.7,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
          }
        }
      >
        Refueling
      </Text>
      <Text
        style={
          {
            "color": "#6a6f71",
            "marginVertical": 13.34,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View>
          <Image
            source={
              {
                "testUri": "../../../app/assets/icons/general/meter.png",
              }
            }
            style={
              [
                {
                  "resizeMode": "contain",
                },
                {
                  "tintColor": "#6a6f71",
                },
                {
                  "height": 30.13699392404255,
                  "width": 30.13699392404255,
                },
                undefined,
              ]
            }
          />
        </View>
        <Text
          style={
            {
              "color": "#6a6f71",
              "marginLeft": 15,
            }
          }
        >
          undefined mi/l
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#ffffff",
      }
    }
  >
    $ 
  </Text>
</View>
`;
